### PR TITLE
feat: adds logflare drain

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Adds a transport to the pino logger library to
forward your messages to Logflare.

This can be verified locally by adding a `.env` file
with the following values:

```
REACT_APP_LOGFLARE_KEY=<LOG_FLARE_API_KEY>
REACT_APP_LOGFLARE_SOURCE=<LOGFLARE_SOURCE_ID>
```

This is a trial with Logflare, we can easily switch
to another provider at a later date.

See attached screen grab for an example of how a message appears in [Logflare](https://logflare.app/guides/overview)
Screenshot 2021-12-28 at 21 02 47

ℹ️ Before merging this change ensure that environmental variables `REACT_APP_LOGFLARE_KEY` and `REACT_APP_LOGFLARE_SOURCE` have been added for all of the instances we wish to add logging to.

### CODE REVIEW

File: src/components/LocationSearch/LocationSearch.tsx [hunk 1]
```diff
@@ -105,14 +105,18 @@ export class LocationSearch extends React.Component<IProps, IState> {
...
     })
     // need to manually trigger an input event as this is what algolia places uses to pick up change
     const event = new Event('input', { bubbles: true })
     this.placesInputRef.current.dispatchEvent(event)
 
+    logger.info('LocationSearch input changed', {
+      locationSearchVariant: this.props?.trackingCategory || 'LocationSearch',
+      value,
+    })
+
```
Comment (thisislawatts) [inline src/components/LocationSearch/LocationSearch.tsx:119]: Introduced a log sibling to the ReactGA.event. This is pretty much only to validate the approach, although I am curious to see if this helps us catch events which will not reach Google Analytics due to users with ad blockers.

Comment (chrismclarke) [reply src/components/LocationSearch/LocationSearch.tsx:119]: Nice to have a test case for sure

File: src/logger/index.ts [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+import { createPinoBrowserSend, createWriteStream } from 'pino-logflare'
 import { getConfigirationOption } from 'src/config/config';
 
-const logLevel = getConfigirationOption('REACT_APP_LOG_LEVEL', 'warn');
+const logLevel = getConfigirationOption('REACT_APP_LOG_LEVEL', 'info');
 
-export const logger = Logger({ browser: { asObject: false }, level: logLevel })
+let loggerInstance;
+
+if (getConfigirationOption('REACT_APP_LOGFLARE_KEY', '')) {
```
Comment (chrismclarke) [inline src/logger/index.ts:9]: So as I understand we can still merge and then just update env later as required (the if statement will catch if no configuration provided?)... If yes I'd say good to go

**Review by chrismclarke — APPROVED** [review]
Nice to include, particularly interesting timing given that logflare has just been acquired by supabase who we plan to migrate our DB to (so mid-term should be a nice shared ecosystem) - and assume until then we can probably qualify for a free tier somewhere. 

I expect there should be a reasonable migration process that we'll just need to do as/when required, also good that they're committed to making it fully open source so that fits well with the overall model.

https://supabase.com/blog/2021/12/02/supabase-acquires-logflare#supabase--logs

**Pull request merged**